src/dataflow/components/dataflow-program.tsx

```diff
@@ -819,6 +839,11 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     return this.getNodeCount("Relay") > 0;
   }
 
+  private isValidRelay(id: string) {
+    // placeholder for more complete validation
+    return true;
+  }
```

Comment:
There's probably going to be some overlap here with the code that I have ready on the `valid-relays-sensors` branch.

Reply:
That's why I stopped at adding the placeholder function.

Reply:
Sounds good.  I'll have a PR ready for mine shortly and we can move forward after that.